src/Compilers/CSharp/Portable/Symbols/Synthesized/Extensions/SynthesizedExtensionMarker.cs

```diff
@@ -0,0 +1,155 @@
+        private readonly ImmutableArray<ParameterSymbol> _parameters;
+
+        internal SynthesizedExtensionMarker(SourceExtensionTypeSymbol extensionType,
+            TypeSymbol underlyingType, ImmutableArray<NamedTypeSymbol> baseExtensionTypes, BindingDiagnosticBag diagnostics)
+        {
+            _extensionType = extensionType;
+            _returnType = Binder.GetSpecialType(DeclaringCompilation, SpecialType.System_Void, extensionType.GetFirstLocation(), diagnostics);
+
+            var parameters = ArrayBuilder<ParameterSymbol>.GetInstance(baseExtensionTypes.Length);
+            parameters.Add(makeParameter(0, underlyingType));
+            for (int i = 0; i < baseExtensionTypes.Length; i++)
+            {
+                parameters.Add(makeParameter(i + 1, baseExtensionTypes[i]));
+            }
+
+            _parameters = parameters.ToImmutableAndFree();
+
+            return;
+
+            ParameterSymbol makeParameter(int ordinal, TypeSymbol parameterType)
+            {
+                return SynthesizedParameterSymbol.Create(container: this,
+                    TypeWithAnnotations.Create(isNullableEnabled: false, parameterType),
+                    ordinal, RefKind.None);
+            }
+        }
+
+        public override string Name => WellKnownMemberNames.ExtensionMarkerMethodName;
+
+        internal override System.Reflection.MethodImplAttributes ImplementationAttributes => default;
+
+        public override bool IsVararg => false;
+
+        public override ImmutableArray<TypeParameterSymbol> TypeParameters => ImmutableArray<TypeParameterSymbol>.Empty;
+
+        internal override int ParameterCount => _parameters.Length;
+
+        public override ImmutableArray<ParameterSymbol> Parameters => _parameters;
+
+        public override RefKind RefKind => RefKind.None;
+
+        public override TypeWithAnnotations ReturnTypeWithAnnotations => TypeWithAnnotations.Create(_returnType);
+
+        public override FlowAnalysisAnnotations ReturnTypeFlowAnalysisAnnotations => FlowAnalysisAnnotations.None;
+
+        public override ImmutableHashSet<string> ReturnNotNullIfParameterNotNull => ImmutableHashSet<string>.Empty;
+
+        internal sealed override bool GenerateDebugInfo => false;
+
+        public override MethodKind MethodKind => MethodKind.Ordinary;
+
+        public override Accessibility DeclaredAccessibility => Accessibility.Private;
+
+        public override bool IsStatic => true;
+
+        public override bool IsVirtual => false;
+
+        public override bool IsOverride => false;
+
+        public override bool IsAbstract => false;
+
+        public override bool IsSealed => false;
+
+        public override bool IsExtern => false;
+
+        internal override bool IsMetadataFinal => false;
+
+        internal override bool IsInitOnly => false;
+
+        public override int Arity => 0;
+
+        public override bool IsExtensionMethod => false;
+
+        internal override bool HasSpecialName => true;
```

Comment:
>HasSpecialName [](http://example.com/codeflow?start=31&length=14)

Any specific reason to return true from this property?

Reply:
It doesn't look like we are paying attention to this on import. Perhaps we should? Also, if we expect runtime to specially recognize the method, we might want to mark it as HasRuntimeSpecialName.

Reply:
I'll punt that question as it's not essential to figure this out now. Added PROTOTYPE